{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "36bb4e87",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from copy import deepcopy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "eec04317",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>condition</th>\n",
       "      <th>avg_lfc</th>\n",
       "      <th>n_obs</th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>Target Gene ID</th>\n",
       "      <th>Category</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>On-Target Ruleset</th>\n",
       "      <th>On-Target Efficacy Score</th>\n",
       "      <th>z_scored_avg_lfc</th>\n",
       "      <th>Domain</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>Strand of sgRNA</th>\n",
       "      <th>Orientation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAAAAAAAAAAGGGCAGAAG</td>\n",
       "      <td>A549</td>\n",
       "      <td>-0.376159</td>\n",
       "      <td>2</td>\n",
       "      <td>APOBEC1</td>\n",
       "      <td>339.0</td>\n",
       "      <td>Non-essential</td>\n",
       "      <td>7665908.0</td>\n",
       "      <td>-999.0</td>\n",
       "      <td>AAAAAAAAAAAAAAAGGGCAGAAGTGGACT</td>\n",
       "      <td>RS3seq-Chen2013</td>\n",
       "      <td>-0.7248</td>\n",
       "      <td>-1.389004</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>12.0</td>\n",
       "      <td>7666908.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>sense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAAAAAAAAAATTAGACCTC</td>\n",
       "      <td>A549</td>\n",
       "      <td>-0.021644</td>\n",
       "      <td>2</td>\n",
       "      <td>MRGPRD</td>\n",
       "      <td>116512.0</td>\n",
       "      <td>Non-essential</td>\n",
       "      <td>68980986.0</td>\n",
       "      <td>-735.0</td>\n",
       "      <td>AAAAAAAAAAAAAAATTAGACCTCAGGGCA</td>\n",
       "      <td>RS3seq-Chen2013</td>\n",
       "      <td>-0.6309</td>\n",
       "      <td>-0.528140</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>68981722.0</td>\n",
       "      <td>-</td>\n",
       "      <td>+</td>\n",
       "      <td>antisense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAAAAAAAAATTAGACCTC</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>-0.257163</td>\n",
       "      <td>2</td>\n",
       "      <td>MRGPRD</td>\n",
       "      <td>116512.0</td>\n",
       "      <td>Non-essential</td>\n",
       "      <td>68980986.0</td>\n",
       "      <td>-735.0</td>\n",
       "      <td>AAAAAAAAAAAAAAATTAGACCTCAGGGCA</td>\n",
       "      <td>RS3seq-Chen2013</td>\n",
       "      <td>-0.6309</td>\n",
       "      <td>-1.381448</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>68981722.0</td>\n",
       "      <td>-</td>\n",
       "      <td>+</td>\n",
       "      <td>antisense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAAAAAAAAATTAGACCTC</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>-0.277625</td>\n",
       "      <td>2</td>\n",
       "      <td>MRGPRD</td>\n",
       "      <td>116512.0</td>\n",
       "      <td>Non-essential</td>\n",
       "      <td>68980986.0</td>\n",
       "      <td>-735.0</td>\n",
       "      <td>AAAAAAAAAAAAAAATTAGACCTCAGGGCA</td>\n",
       "      <td>RS3seq-Chen2013</td>\n",
       "      <td>-0.6309</td>\n",
       "      <td>-1.315569</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>11.0</td>\n",
       "      <td>68981722.0</td>\n",
       "      <td>-</td>\n",
       "      <td>+</td>\n",
       "      <td>antisense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAAAAAAAACAGGACACAG</td>\n",
       "      <td>A549</td>\n",
       "      <td>0.031810</td>\n",
       "      <td>2</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>23395.0</td>\n",
       "      <td>Essential</td>\n",
       "      <td>45388577.0</td>\n",
       "      <td>667.0</td>\n",
       "      <td>CAGAAAAAAAAAAACAGGACACAGGGGAGA</td>\n",
       "      <td>RS3seq-Chen2013</td>\n",
       "      <td>0.0140</td>\n",
       "      <td>-0.398339</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>45389244.0</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>antisense</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         sgRNA Sequence condition   avg_lfc  n_obs Target Gene Symbol  \\\n",
       "0  AAAAAAAAAAAGGGCAGAAG      A549 -0.376159      2            APOBEC1   \n",
       "1  AAAAAAAAAAATTAGACCTC      A549 -0.021644      2             MRGPRD   \n",
       "2  AAAAAAAAAAATTAGACCTC    HCT116 -0.257163      2             MRGPRD   \n",
       "3  AAAAAAAAAAATTAGACCTC    HCT116 -0.277625      2             MRGPRD   \n",
       "4  AAAAAAAAAACAGGACACAG      A549  0.031810      2              LARS2   \n",
       "\n",
       "   Target Gene ID       Category  TSS Position  sgRNA 'Cut' Site TSS Offset  \\\n",
       "0           339.0  Non-essential     7665908.0                       -999.0   \n",
       "1        116512.0  Non-essential    68980986.0                       -735.0   \n",
       "2        116512.0  Non-essential    68980986.0                       -735.0   \n",
       "3        116512.0  Non-essential    68980986.0                       -735.0   \n",
       "4         23395.0      Essential    45388577.0                        667.0   \n",
       "\n",
       "           sgRNA Context Sequence On-Target Ruleset  On-Target Efficacy Score  \\\n",
       "0  AAAAAAAAAAAAAAAGGGCAGAAGTGGACT   RS3seq-Chen2013                   -0.7248   \n",
       "1  AAAAAAAAAAAAAAATTAGACCTCAGGGCA   RS3seq-Chen2013                   -0.6309   \n",
       "2  AAAAAAAAAAAAAAATTAGACCTCAGGGCA   RS3seq-Chen2013                   -0.6309   \n",
       "3  AAAAAAAAAAAAAAATTAGACCTCAGGGCA   RS3seq-Chen2013                   -0.6309   \n",
       "4  CAGAAAAAAAAAAACAGGACACAGGGGAGA   RS3seq-Chen2013                    0.0140   \n",
       "\n",
       "   z_scored_avg_lfc Domain  chromosome  sgRNA 'Cut' Position Strand of Target  \\\n",
       "0         -1.389004   Kox1        12.0             7666908.0                -   \n",
       "1         -0.528140   Kox1        11.0            68981722.0                -   \n",
       "2         -1.381448   Kox1        11.0            68981722.0                -   \n",
       "3         -1.315569   Zim3        11.0            68981722.0                -   \n",
       "4         -0.398339   Kox1         3.0            45389244.0                +   \n",
       "\n",
       "  Strand of sgRNA Orientation  \n",
       "0               -       sense  \n",
       "1               +   antisense  \n",
       "2               +   antisense  \n",
       "3               +   antisense  \n",
       "4               -   antisense  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in processed in-house datasets, subset to targeting guides (i.e. all guides except intergenic and non-targeting controls)\n",
    "file_path = '../../Data/primaryLibrary/'\n",
    "annot_guide_lfcs_A549_Kox1 = pd.read_csv(file_path + 'A549_Kox1_zscored.csv')\n",
    "annot_guide_lfcs_A549_Kox1 = annot_guide_lfcs_A549_Kox1[annot_guide_lfcs_A549_Kox1.Category.isin(['Essential', 'Non-essential'])]\n",
    "\n",
    "annot_guide_lfcs_A549_Zim3 = pd.read_csv(file_path +  'A549_Zim3_zscored.csv')\n",
    "annot_guide_lfcs_A549_Zim3 = annot_guide_lfcs_A549_Zim3[annot_guide_lfcs_A549_Zim3.Category.isin(['Essential', 'Non-essential'])]\n",
    "\n",
    "annot_guide_lfcs_HCT116_Kox1 = pd.read_csv(file_path +  'HCT116_Kox1_zscored.csv')\n",
    "annot_guide_lfcs_HCT116_Kox1 = annot_guide_lfcs_HCT116_Kox1[annot_guide_lfcs_HCT116_Kox1.Category.isin(['Essential', 'Non-essential'])]\n",
    "\n",
    "annot_guide_lfcs_HCT116_Zim3 = pd.read_csv(file_path +  'HCT116_Zim3_zscored.csv')\n",
    "annot_guide_lfcs_HCT116_Zim3 = annot_guide_lfcs_HCT116_Zim3[annot_guide_lfcs_HCT116_Zim3.Category.isin(['Essential', 'Non-essential'])]\n",
    "\n",
    "#add column to specify domain before concatenating\n",
    "annot_guide_lfcs_A549_Kox1['Domain'] = 'Kox1'\n",
    "annot_guide_lfcs_A549_Zim3['Domain'] = 'Zim3'\n",
    "annot_guide_lfcs_HCT116_Kox1['Domain'] = 'Kox1'\n",
    "annot_guide_lfcs_HCT116_Zim3['Domain'] = 'Zim3'\n",
    "\n",
    "InHouseCleanData = pd.concat([annot_guide_lfcs_A549_Kox1, annot_guide_lfcs_A549_Zim3, annot_guide_lfcs_HCT116_Kox1, annot_guide_lfcs_HCT116_Zim3])\n",
    "\n",
    "# read in design file to append additional features\n",
    "ess_noness = pd.read_csv(file_path + 'essentials_nonessentials_CRISPRi_tiling_designs_v2.csv')\n",
    "ess_noness['chromosome'] = ess_noness['Reference Sequence'].apply(lambda x: int(x.split(\"NC_\")[1].split(\".\")[0]) if pd.notnull(x) else x)\n",
    "ess_noness = ess_noness[['Target Gene Symbol','Target Gene ID','Category','chromosome','sgRNA Sequence', \n",
    "           'sgRNA Context Sequence', 'TSS Position', \n",
    "       'sgRNA \\'Cut\\' Site TSS Offset', 'sgRNA \\'Cut\\' Position', \n",
    "            'Strand of Target','Strand of sgRNA', 'Orientation',\n",
    "            'On-Target Ruleset','On-Target Efficacy Score']]\n",
    "InHouseCleanData = pd.merge(InHouseCleanData, ess_noness)\n",
    "\n",
    "InHouseCleanData.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a43a7615-bc0e-4ae5-95e9-e6250c17f53f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>strand targeted</th>\n",
       "      <th>PAM genomic coordinate [hg38]</th>\n",
       "      <th>context seq</th>\n",
       "      <th>guide id</th>\n",
       "      <th>oligo</th>\n",
       "      <th>sgRNA sequence</th>\n",
       "      <th>zscoreRelativeNC</th>\n",
       "      <th>Avg_LFC</th>\n",
       "      <th>sgRNA sequence in Context</th>\n",
       "      <th>Target Gene Function</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118567638.0</td>\n",
       "      <td>GCACTTTCCTTGTTTACCTCTGGTAGGTTT</td>\n",
       "      <td>ARCN1_w_118438354.27</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCACTTTCCTTGTTTACCTCTGGT...</td>\n",
       "      <td>GCACTTTCCTTGTTTACCTCTGGT</td>\n",
       "      <td>0.250358</td>\n",
       "      <td>-0.134928</td>\n",
       "      <td>True</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>-0.250358</td>\n",
       "      <td>0.134928</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568026.0</td>\n",
       "      <td>GCAACCCTGTAAGCATGCTTCTTGAGGAGT</td>\n",
       "      <td>ARCN1_w_118438742.27</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCAACCCTGTAAGCATGCTTCTTG...</td>\n",
       "      <td>GCAACCCTGTAAGCATGCTTCTTG</td>\n",
       "      <td>0.157641</td>\n",
       "      <td>-0.166789</td>\n",
       "      <td>True</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>-0.157641</td>\n",
       "      <td>0.166789</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568080.0</td>\n",
       "      <td>GATGCTGCTTCTTCGGAAATATAACGGTAG</td>\n",
       "      <td>ARCN1_w_118438796.24</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCTGCTTCTTCGGAAATATAAGTT...</td>\n",
       "      <td>GCTGCTTCTTCGGAAATATAA</td>\n",
       "      <td>0.216340</td>\n",
       "      <td>-0.146618</td>\n",
       "      <td>True</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>-0.216340</td>\n",
       "      <td>0.146618</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568134.0</td>\n",
       "      <td>TAGCACCTTCCAGGGCTTCCAACTTGGATC</td>\n",
       "      <td>ARCN1_w_118438850.25</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCACCTTCCAGGGCTTCCAACTGT...</td>\n",
       "      <td>GCACCTTCCAGGGCTTCCAACT</td>\n",
       "      <td>0.320467</td>\n",
       "      <td>-0.110835</td>\n",
       "      <td>True</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>-0.320467</td>\n",
       "      <td>0.110835</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568146.0</td>\n",
       "      <td>ATTGCCTTATTGTAGCACCTTCCAGGGCTT</td>\n",
       "      <td>ARCN1_w_118438862.24</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCCTTATTGTAGCACCTTCCAGTT...</td>\n",
       "      <td>GCCTTATTGTAGCACCTTCCA</td>\n",
       "      <td>-0.708019</td>\n",
       "      <td>-0.464269</td>\n",
       "      <td>True</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>0.708019</td>\n",
       "      <td>0.464269</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Target Gene Symbol chromosome strand targeted  \\\n",
       "0              ARCN1      chr11               +   \n",
       "1              ARCN1      chr11               +   \n",
       "2              ARCN1      chr11               +   \n",
       "3              ARCN1      chr11               +   \n",
       "4              ARCN1      chr11               +   \n",
       "\n",
       "   PAM genomic coordinate [hg38]                     context seq  \\\n",
       "0                    118567638.0  GCACTTTCCTTGTTTACCTCTGGTAGGTTT   \n",
       "1                    118568026.0  GCAACCCTGTAAGCATGCTTCTTGAGGAGT   \n",
       "2                    118568080.0  GATGCTGCTTCTTCGGAAATATAACGGTAG   \n",
       "3                    118568134.0  TAGCACCTTCCAGGGCTTCCAACTTGGATC   \n",
       "4                    118568146.0  ATTGCCTTATTGTAGCACCTTCCAGGGCTT   \n",
       "\n",
       "               guide id                                              oligo  \\\n",
       "0  ARCN1_w_118438354.27  cccttggagaaCCAcctTGTTGGCACTTTCCTTGTTTACCTCTGGT...   \n",
       "1  ARCN1_w_118438742.27  cccttggagaaCCAcctTGTTGGCAACCCTGTAAGCATGCTTCTTG...   \n",
       "2  ARCN1_w_118438796.24  cccttggagaaCCAcctTGTTGGCTGCTTCTTCGGAAATATAAGTT...   \n",
       "3  ARCN1_w_118438850.25  cccttggagaaCCAcctTGTTGGCACCTTCCAGGGCTTCCAACTGT...   \n",
       "4  ARCN1_w_118438862.24  cccttggagaaCCAcctTGTTGGCCTTATTGTAGCACCTTCCAGTT...   \n",
       "\n",
       "             sgRNA sequence  zscoreRelativeNC   Avg_LFC  \\\n",
       "0  GCACTTTCCTTGTTTACCTCTGGT          0.250358 -0.134928   \n",
       "1  GCAACCCTGTAAGCATGCTTCTTG          0.157641 -0.166789   \n",
       "2     GCTGCTTCTTCGGAAATATAA          0.216340 -0.146618   \n",
       "3    GCACCTTCCAGGGCTTCCAACT          0.320467 -0.110835   \n",
       "4     GCCTTATTGTAGCACCTTCCA         -0.708019 -0.464269   \n",
       "\n",
       "   sgRNA sequence in Context Target Gene Function  zscoreRelativeNC_signed  \\\n",
       "0                       True           Resistance                -0.250358   \n",
       "1                       True           Resistance                -0.157641   \n",
       "2                       True           Resistance                -0.216340   \n",
       "3                       True           Resistance                -0.320467   \n",
       "4                       True           Resistance                 0.708019   \n",
       "\n",
       "   Avg_LFC_signed condition Domain  \n",
       "0        0.134928      K562   Kox1  \n",
       "1        0.166789      K562   Kox1  \n",
       "2        0.146618      K562   Kox1  \n",
       "3        0.110835      K562   Kox1  \n",
       "4        0.464269      K562   Kox1  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in cleaned Gilbert ricin data\n",
    "file_path = '../../Data/externalData/'\n",
    "GilbertRicinCleanData = pd.read_csv(file_path + 'Gilbert2014Cleaned.csv')\n",
    "GilbertRicinCleanData['condition'] = 'K562'\n",
    "GilbertRicinCleanData['Domain'] = 'Kox1'\n",
    "GilbertRicinCleanData.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "bca23ad4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>strand</th>\n",
       "      <th>sequence</th>\n",
       "      <th>context_seq</th>\n",
       "      <th>hg38_coord</th>\n",
       "      <th>start_coord_hg38</th>\n",
       "      <th>end_coord_hg38</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTR6_+_100592059</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GCGAGCACATAGAGGGATCT</td>\n",
       "      <td>TGCTTCGAGCACATAGAGGGATCTGGGCCC</td>\n",
       "      <td>100198281</td>\n",
       "      <td>100198284</td>\n",
       "      <td>100198303</td>\n",
       "      <td>-0.018881</td>\n",
       "      <td>0.035104</td>\n",
       "      <td>-0.889828</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACTR6_+_100592060</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GTCGAGCACATAGAGGGATC</td>\n",
       "      <td>CTGCTTCGAGCACATAGAGGGATCTGGGCC</td>\n",
       "      <td>100198282</td>\n",
       "      <td>100198285</td>\n",
       "      <td>100198304</td>\n",
       "      <td>0.009292</td>\n",
       "      <td>0.295731</td>\n",
       "      <td>0.580559</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTR6_+_100592066</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GCCTGCTTCGAGCACATAGA</td>\n",
       "      <td>GTCCACCTGCTTCGAGCACATAGAGGGATC</td>\n",
       "      <td>100198288</td>\n",
       "      <td>100198291</td>\n",
       "      <td>100198310</td>\n",
       "      <td>0.000734</td>\n",
       "      <td>0.214023</td>\n",
       "      <td>0.119582</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTR6_+_100592067</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GACCTGCTTCGAGCACATAG</td>\n",
       "      <td>GGTCCACCTGCTTCGAGCACATAGAGGGAT</td>\n",
       "      <td>100198289</td>\n",
       "      <td>100198292</td>\n",
       "      <td>100198311</td>\n",
       "      <td>-0.008256</td>\n",
       "      <td>0.131427</td>\n",
       "      <td>-0.346402</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTR6_+_100592092</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GAGACTCTGTCACCTCCGAG</td>\n",
       "      <td>GGGCTAGACTCTGTCACCTCCGAGGGGTCC</td>\n",
       "      <td>100198314</td>\n",
       "      <td>100198317</td>\n",
       "      <td>100198336</td>\n",
       "      <td>0.001314</td>\n",
       "      <td>0.165469</td>\n",
       "      <td>-0.154343</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  ID Target Gene Symbol chromosome strand  \\\n",
       "0  ACTR6_+_100592059              ACTR6      chr12      +   \n",
       "1  ACTR6_+_100592060              ACTR6      chr12      +   \n",
       "2  ACTR6_+_100592066              ACTR6      chr12      +   \n",
       "3  ACTR6_+_100592067              ACTR6      chr12      +   \n",
       "4  ACTR6_+_100592092              ACTR6      chr12      +   \n",
       "\n",
       "               sequence                     context_seq  hg38_coord  \\\n",
       "0  GCGAGCACATAGAGGGATCT  TGCTTCGAGCACATAGAGGGATCTGGGCCC   100198281   \n",
       "1  GTCGAGCACATAGAGGGATC  CTGCTTCGAGCACATAGAGGGATCTGGGCC   100198282   \n",
       "2  GCCTGCTTCGAGCACATAGA  GTCCACCTGCTTCGAGCACATAGAGGGATC   100198288   \n",
       "3  GACCTGCTTCGAGCACATAG  GGTCCACCTGCTTCGAGCACATAGAGGGAT   100198289   \n",
       "4  GAGACTCTGTCACCTCCGAG  GGGCTAGACTCTGTCACCTCCGAGGGGTCC   100198314   \n",
       "\n",
       "   start_coord_hg38  end_coord_hg38  Phenotype scores-ave_Rep1_Rep2  \\\n",
       "0         100198284       100198303                       -0.018881   \n",
       "1         100198285       100198304                        0.009292   \n",
       "2         100198291       100198310                        0.000734   \n",
       "3         100198292       100198311                       -0.008256   \n",
       "4         100198317       100198336                        0.001314   \n",
       "\n",
       "   Avg_LFC-Tfinal  z-score relative to NC condition Domain  \n",
       "0        0.035104               -0.889828      K562   Kox1  \n",
       "1        0.295731                0.580559      K562   Kox1  \n",
       "2        0.214023                0.119582      K562   Kox1  \n",
       "3        0.131427               -0.346402      K562   Kox1  \n",
       "4        0.165469               -0.154343      K562   Kox1  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in cleaned Nunez data\n",
    "NunezCleanData = pd.read_csv(file_path + 'Nunez2021Cleaned.csv')\n",
    "NunezCleanData['condition'] = 'K562'\n",
    "NunezCleanData['Domain'] = 'Kox1'\n",
    "NunezCleanData.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "3fdb116a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# re-assigns sgRNA \"cut\" site in Gilbert and Nunez Datasets to be consistent with CRISPick\n",
    "def CRISPickCutSite(df, strand_col, point_col, context_seq_col):\n",
    "    CutSiteList = []\n",
    "    for index, row in df.iterrows():\n",
    "        if row[strand_col] == '+':\n",
    "            CutSiteList.append(row[point_col] + 7)\n",
    "        elif row[strand_col] == '-':\n",
    "            CutSiteList.append(row[point_col] - 4)\n",
    "        else:\n",
    "            print(row[context_seq_col] + f'strand is neither + nor -')\n",
    "    return CutSiteList"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "56ef71de",
   "metadata": {},
   "outputs": [],
   "source": [
    "geneListGilbertAndNunez = GilbertRicinCleanData['Target Gene Symbol'].unique().tolist() + NunezCleanData['Target Gene Symbol'].unique().tolist()\n",
    "pd.DataFrame(geneListGilbertAndNunez).to_csv(file_path + 'geneListGilbertAndNunez.csv', index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b02f5b5-3afa-42c5-a463-469850388838",
   "metadata": {},
   "source": [
    "Run CRISPick providing this list of genes as input to obtain MANE Select TSS positions. Link to output directory: https://portals.broadinstitute.org/gppx/crispick/platform/results/1da3859a-902d-471a-b06b-7aa43da31abd."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "e4326095-7dec-42f8-a48b-7c080ae70423",
   "metadata": {},
   "outputs": [],
   "source": [
    "CRISPick_geneListGilbertAndNunez = pd.read_table(file_path + 'geneListGilbertAndNunez-sgrna-designs.txt')\n",
    "CRISPick_geneListGilbertAndNunez = CRISPick_geneListGilbertAndNunez[[ \n",
    "       'Target Gene Symbol',  'Reference Sequence', 'Strand of Target',\n",
    "       'TSS Position']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c24742d1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>strand targeted</th>\n",
       "      <th>PAM genomic coordinate [hg38]</th>\n",
       "      <th>context seq</th>\n",
       "      <th>guide id</th>\n",
       "      <th>oligo</th>\n",
       "      <th>sgRNA sequence</th>\n",
       "      <th>zscoreRelativeNC</th>\n",
       "      <th>Avg_LFC</th>\n",
       "      <th>sgRNA sequence in Context</th>\n",
       "      <th>Target Gene Function</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>CRISPick CutSite</th>\n",
       "      <th>Reference Sequence</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118567638.0</td>\n",
       "      <td>GCACTTTCCTTGTTTACCTCTGGTAGGTTT</td>\n",
       "      <td>ARCN1_w_118438354.27</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCACTTTCCTTGTTTACCTCTGGT...</td>\n",
       "      <td>GCACTTTCCTTGTTTACCTCTGGT</td>\n",
       "      <td>0.250358</td>\n",
       "      <td>-0.134928</td>\n",
       "      <td>True</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>-0.250358</td>\n",
       "      <td>0.134928</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118567645.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568026.0</td>\n",
       "      <td>GCAACCCTGTAAGCATGCTTCTTGAGGAGT</td>\n",
       "      <td>ARCN1_w_118438742.27</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCAACCCTGTAAGCATGCTTCTTG...</td>\n",
       "      <td>GCAACCCTGTAAGCATGCTTCTTG</td>\n",
       "      <td>0.157641</td>\n",
       "      <td>-0.166789</td>\n",
       "      <td>True</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>-0.157641</td>\n",
       "      <td>0.166789</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568033.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568080.0</td>\n",
       "      <td>GATGCTGCTTCTTCGGAAATATAACGGTAG</td>\n",
       "      <td>ARCN1_w_118438796.24</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCTGCTTCTTCGGAAATATAAGTT...</td>\n",
       "      <td>GCTGCTTCTTCGGAAATATAA</td>\n",
       "      <td>0.216340</td>\n",
       "      <td>-0.146618</td>\n",
       "      <td>True</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>-0.216340</td>\n",
       "      <td>0.146618</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568087.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568134.0</td>\n",
       "      <td>TAGCACCTTCCAGGGCTTCCAACTTGGATC</td>\n",
       "      <td>ARCN1_w_118438850.25</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCACCTTCCAGGGCTTCCAACTGT...</td>\n",
       "      <td>GCACCTTCCAGGGCTTCCAACT</td>\n",
       "      <td>0.320467</td>\n",
       "      <td>-0.110835</td>\n",
       "      <td>True</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>-0.320467</td>\n",
       "      <td>0.110835</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568141.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568146.0</td>\n",
       "      <td>ATTGCCTTATTGTAGCACCTTCCAGGGCTT</td>\n",
       "      <td>ARCN1_w_118438862.24</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCCTTATTGTAGCACCTTCCAGTT...</td>\n",
       "      <td>GCCTTATTGTAGCACCTTCCA</td>\n",
       "      <td>-0.708019</td>\n",
       "      <td>-0.464269</td>\n",
       "      <td>True</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>0.708019</td>\n",
       "      <td>0.464269</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568153.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Target Gene Symbol chromosome strand targeted  \\\n",
       "0              ARCN1      chr11               +   \n",
       "1              ARCN1      chr11               +   \n",
       "2              ARCN1      chr11               +   \n",
       "3              ARCN1      chr11               +   \n",
       "4              ARCN1      chr11               +   \n",
       "\n",
       "   PAM genomic coordinate [hg38]                     context seq  \\\n",
       "0                    118567638.0  GCACTTTCCTTGTTTACCTCTGGTAGGTTT   \n",
       "1                    118568026.0  GCAACCCTGTAAGCATGCTTCTTGAGGAGT   \n",
       "2                    118568080.0  GATGCTGCTTCTTCGGAAATATAACGGTAG   \n",
       "3                    118568134.0  TAGCACCTTCCAGGGCTTCCAACTTGGATC   \n",
       "4                    118568146.0  ATTGCCTTATTGTAGCACCTTCCAGGGCTT   \n",
       "\n",
       "               guide id                                              oligo  \\\n",
       "0  ARCN1_w_118438354.27  cccttggagaaCCAcctTGTTGGCACTTTCCTTGTTTACCTCTGGT...   \n",
       "1  ARCN1_w_118438742.27  cccttggagaaCCAcctTGTTGGCAACCCTGTAAGCATGCTTCTTG...   \n",
       "2  ARCN1_w_118438796.24  cccttggagaaCCAcctTGTTGGCTGCTTCTTCGGAAATATAAGTT...   \n",
       "3  ARCN1_w_118438850.25  cccttggagaaCCAcctTGTTGGCACCTTCCAGGGCTTCCAACTGT...   \n",
       "4  ARCN1_w_118438862.24  cccttggagaaCCAcctTGTTGGCCTTATTGTAGCACCTTCCAGTT...   \n",
       "\n",
       "             sgRNA sequence  zscoreRelativeNC   Avg_LFC  \\\n",
       "0  GCACTTTCCTTGTTTACCTCTGGT          0.250358 -0.134928   \n",
       "1  GCAACCCTGTAAGCATGCTTCTTG          0.157641 -0.166789   \n",
       "2     GCTGCTTCTTCGGAAATATAA          0.216340 -0.146618   \n",
       "3    GCACCTTCCAGGGCTTCCAACT          0.320467 -0.110835   \n",
       "4     GCCTTATTGTAGCACCTTCCA         -0.708019 -0.464269   \n",
       "\n",
       "   sgRNA sequence in Context Target Gene Function  zscoreRelativeNC_signed  \\\n",
       "0                       True           Resistance                -0.250358   \n",
       "1                       True           Resistance                -0.157641   \n",
       "2                       True           Resistance                -0.216340   \n",
       "3                       True           Resistance                -0.320467   \n",
       "4                       True           Resistance                 0.708019   \n",
       "\n",
       "   Avg_LFC_signed condition Domain  CRISPick CutSite Reference Sequence  \\\n",
       "0        0.134928      K562   Kox1       118567645.0       NC_000011.10   \n",
       "1        0.166789      K562   Kox1       118568033.0       NC_000011.10   \n",
       "2        0.146618      K562   Kox1       118568087.0       NC_000011.10   \n",
       "3        0.110835      K562   Kox1       118568141.0       NC_000011.10   \n",
       "4        0.464269      K562   Kox1       118568153.0       NC_000011.10   \n",
       "\n",
       "  Strand of Target  TSS Position  \n",
       "0                +     118572410  \n",
       "1                +     118572410  \n",
       "2                +     118572410  \n",
       "3                +     118572410  \n",
       "4                +     118572410  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#reassign cut site for Gilbert dataset guides\n",
    "GilbertRicinCleanDataCutSite = CRISPickCutSite(GilbertRicinCleanData,'strand targeted' ,'PAM genomic coordinate [hg38]', 'context seq')\n",
    "GilbertRicinCleanData['CRISPick CutSite'] = GilbertRicinCleanDataCutSite\n",
    "GilbertRicinCleanData = pd.merge(GilbertRicinCleanData,CRISPick_geneListGilbertAndNunez, \n",
    "                                                 on = 'Target Gene Symbol')\n",
    "GilbertRicinCleanData.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4fbd09cf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>strand targeted</th>\n",
       "      <th>PAM genomic coordinate [hg38]</th>\n",
       "      <th>context seq</th>\n",
       "      <th>guide id</th>\n",
       "      <th>oligo</th>\n",
       "      <th>sgRNA sequence</th>\n",
       "      <th>zscoreRelativeNC</th>\n",
       "      <th>Avg_LFC</th>\n",
       "      <th>...</th>\n",
       "      <th>Target Gene Function</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>CRISPick CutSite</th>\n",
       "      <th>Reference Sequence</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>Distance to TSS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118567638.0</td>\n",
       "      <td>GCACTTTCCTTGTTTACCTCTGGTAGGTTT</td>\n",
       "      <td>ARCN1_w_118438354.27</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCACTTTCCTTGTTTACCTCTGGT...</td>\n",
       "      <td>GCACTTTCCTTGTTTACCTCTGGT</td>\n",
       "      <td>0.250358</td>\n",
       "      <td>-0.134928</td>\n",
       "      <td>...</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>-0.250358</td>\n",
       "      <td>0.134928</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118567645.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "      <td>-4765.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568026.0</td>\n",
       "      <td>GCAACCCTGTAAGCATGCTTCTTGAGGAGT</td>\n",
       "      <td>ARCN1_w_118438742.27</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCAACCCTGTAAGCATGCTTCTTG...</td>\n",
       "      <td>GCAACCCTGTAAGCATGCTTCTTG</td>\n",
       "      <td>0.157641</td>\n",
       "      <td>-0.166789</td>\n",
       "      <td>...</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>-0.157641</td>\n",
       "      <td>0.166789</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568033.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "      <td>-4377.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568080.0</td>\n",
       "      <td>GATGCTGCTTCTTCGGAAATATAACGGTAG</td>\n",
       "      <td>ARCN1_w_118438796.24</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCTGCTTCTTCGGAAATATAAGTT...</td>\n",
       "      <td>GCTGCTTCTTCGGAAATATAA</td>\n",
       "      <td>0.216340</td>\n",
       "      <td>-0.146618</td>\n",
       "      <td>...</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>-0.216340</td>\n",
       "      <td>0.146618</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568087.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "      <td>-4323.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568134.0</td>\n",
       "      <td>TAGCACCTTCCAGGGCTTCCAACTTGGATC</td>\n",
       "      <td>ARCN1_w_118438850.25</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCACCTTCCAGGGCTTCCAACTGT...</td>\n",
       "      <td>GCACCTTCCAGGGCTTCCAACT</td>\n",
       "      <td>0.320467</td>\n",
       "      <td>-0.110835</td>\n",
       "      <td>...</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>-0.320467</td>\n",
       "      <td>0.110835</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568141.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "      <td>-4269.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568146.0</td>\n",
       "      <td>ATTGCCTTATTGTAGCACCTTCCAGGGCTT</td>\n",
       "      <td>ARCN1_w_118438862.24</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCCTTATTGTAGCACCTTCCAGTT...</td>\n",
       "      <td>GCCTTATTGTAGCACCTTCCA</td>\n",
       "      <td>-0.708019</td>\n",
       "      <td>-0.464269</td>\n",
       "      <td>...</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>0.708019</td>\n",
       "      <td>0.464269</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568153.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "      <td>-4257.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Target Gene Symbol chromosome strand targeted  \\\n",
       "0              ARCN1      chr11               +   \n",
       "1              ARCN1      chr11               +   \n",
       "2              ARCN1      chr11               +   \n",
       "3              ARCN1      chr11               +   \n",
       "4              ARCN1      chr11               +   \n",
       "\n",
       "   PAM genomic coordinate [hg38]                     context seq  \\\n",
       "0                    118567638.0  GCACTTTCCTTGTTTACCTCTGGTAGGTTT   \n",
       "1                    118568026.0  GCAACCCTGTAAGCATGCTTCTTGAGGAGT   \n",
       "2                    118568080.0  GATGCTGCTTCTTCGGAAATATAACGGTAG   \n",
       "3                    118568134.0  TAGCACCTTCCAGGGCTTCCAACTTGGATC   \n",
       "4                    118568146.0  ATTGCCTTATTGTAGCACCTTCCAGGGCTT   \n",
       "\n",
       "               guide id                                              oligo  \\\n",
       "0  ARCN1_w_118438354.27  cccttggagaaCCAcctTGTTGGCACTTTCCTTGTTTACCTCTGGT...   \n",
       "1  ARCN1_w_118438742.27  cccttggagaaCCAcctTGTTGGCAACCCTGTAAGCATGCTTCTTG...   \n",
       "2  ARCN1_w_118438796.24  cccttggagaaCCAcctTGTTGGCTGCTTCTTCGGAAATATAAGTT...   \n",
       "3  ARCN1_w_118438850.25  cccttggagaaCCAcctTGTTGGCACCTTCCAGGGCTTCCAACTGT...   \n",
       "4  ARCN1_w_118438862.24  cccttggagaaCCAcctTGTTGGCCTTATTGTAGCACCTTCCAGTT...   \n",
       "\n",
       "             sgRNA sequence  zscoreRelativeNC   Avg_LFC  ...  \\\n",
       "0  GCACTTTCCTTGTTTACCTCTGGT          0.250358 -0.134928  ...   \n",
       "1  GCAACCCTGTAAGCATGCTTCTTG          0.157641 -0.166789  ...   \n",
       "2     GCTGCTTCTTCGGAAATATAA          0.216340 -0.146618  ...   \n",
       "3    GCACCTTCCAGGGCTTCCAACT          0.320467 -0.110835  ...   \n",
       "4     GCCTTATTGTAGCACCTTCCA         -0.708019 -0.464269  ...   \n",
       "\n",
       "   Target Gene Function zscoreRelativeNC_signed  Avg_LFC_signed  condition  \\\n",
       "0            Resistance               -0.250358        0.134928       K562   \n",
       "1            Resistance               -0.157641        0.166789       K562   \n",
       "2            Resistance               -0.216340        0.146618       K562   \n",
       "3            Resistance               -0.320467        0.110835       K562   \n",
       "4            Resistance                0.708019        0.464269       K562   \n",
       "\n",
       "  Domain CRISPick CutSite  Reference Sequence Strand of Target TSS Position  \\\n",
       "0   Kox1      118567645.0        NC_000011.10                +    118572410   \n",
       "1   Kox1      118568033.0        NC_000011.10                +    118572410   \n",
       "2   Kox1      118568087.0        NC_000011.10                +    118572410   \n",
       "3   Kox1      118568141.0        NC_000011.10                +    118572410   \n",
       "4   Kox1      118568153.0        NC_000011.10                +    118572410   \n",
       "\n",
       "   Distance to TSS  \n",
       "0          -4765.0  \n",
       "1          -4377.0  \n",
       "2          -4323.0  \n",
       "3          -4269.0  \n",
       "4          -4257.0  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#calculate guide distance from MANE Select annotated TSS \n",
    "GilbertRicinCleanData[\"Distance to TSS\"] = GilbertRicinCleanData.apply(\n",
    "    lambda r:(r[\"CRISPick CutSite\"] - r[\"TSS Position\"]) if r[\"Strand of Target\"] == \"+\" \\\n",
    "              else (r[\"TSS Position\"] - r[\"CRISPick CutSite\"]), axis=1)\n",
    "GilbertRicinCleanData = GilbertRicinCleanData[abs(GilbertRicinCleanData['Distance to TSS']) <= 5000]\n",
    "GilbertRicinCleanData.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "2dd65b43",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>strand</th>\n",
       "      <th>sequence</th>\n",
       "      <th>context_seq</th>\n",
       "      <th>hg38_coord</th>\n",
       "      <th>start_coord_hg38</th>\n",
       "      <th>end_coord_hg38</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>CRISPick CutSite</th>\n",
       "      <th>Reference Sequence</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTR6_+_100592059</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GCGAGCACATAGAGGGATCT</td>\n",
       "      <td>TGCTTCGAGCACATAGAGGGATCTGGGCCC</td>\n",
       "      <td>100198281</td>\n",
       "      <td>100198284</td>\n",
       "      <td>100198303</td>\n",
       "      <td>-0.018881</td>\n",
       "      <td>0.035104</td>\n",
       "      <td>-0.889828</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198288</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACTR6_+_100592060</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GTCGAGCACATAGAGGGATC</td>\n",
       "      <td>CTGCTTCGAGCACATAGAGGGATCTGGGCC</td>\n",
       "      <td>100198282</td>\n",
       "      <td>100198285</td>\n",
       "      <td>100198304</td>\n",
       "      <td>0.009292</td>\n",
       "      <td>0.295731</td>\n",
       "      <td>0.580559</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198289</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTR6_+_100592066</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GCCTGCTTCGAGCACATAGA</td>\n",
       "      <td>GTCCACCTGCTTCGAGCACATAGAGGGATC</td>\n",
       "      <td>100198288</td>\n",
       "      <td>100198291</td>\n",
       "      <td>100198310</td>\n",
       "      <td>0.000734</td>\n",
       "      <td>0.214023</td>\n",
       "      <td>0.119582</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198295</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTR6_+_100592067</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GACCTGCTTCGAGCACATAG</td>\n",
       "      <td>GGTCCACCTGCTTCGAGCACATAGAGGGAT</td>\n",
       "      <td>100198289</td>\n",
       "      <td>100198292</td>\n",
       "      <td>100198311</td>\n",
       "      <td>-0.008256</td>\n",
       "      <td>0.131427</td>\n",
       "      <td>-0.346402</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198296</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTR6_+_100592092</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GAGACTCTGTCACCTCCGAG</td>\n",
       "      <td>GGGCTAGACTCTGTCACCTCCGAGGGGTCC</td>\n",
       "      <td>100198314</td>\n",
       "      <td>100198317</td>\n",
       "      <td>100198336</td>\n",
       "      <td>0.001314</td>\n",
       "      <td>0.165469</td>\n",
       "      <td>-0.154343</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198321</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  ID Target Gene Symbol chromosome strand  \\\n",
       "0  ACTR6_+_100592059              ACTR6      chr12      +   \n",
       "1  ACTR6_+_100592060              ACTR6      chr12      +   \n",
       "2  ACTR6_+_100592066              ACTR6      chr12      +   \n",
       "3  ACTR6_+_100592067              ACTR6      chr12      +   \n",
       "4  ACTR6_+_100592092              ACTR6      chr12      +   \n",
       "\n",
       "               sequence                     context_seq  hg38_coord  \\\n",
       "0  GCGAGCACATAGAGGGATCT  TGCTTCGAGCACATAGAGGGATCTGGGCCC   100198281   \n",
       "1  GTCGAGCACATAGAGGGATC  CTGCTTCGAGCACATAGAGGGATCTGGGCC   100198282   \n",
       "2  GCCTGCTTCGAGCACATAGA  GTCCACCTGCTTCGAGCACATAGAGGGATC   100198288   \n",
       "3  GACCTGCTTCGAGCACATAG  GGTCCACCTGCTTCGAGCACATAGAGGGAT   100198289   \n",
       "4  GAGACTCTGTCACCTCCGAG  GGGCTAGACTCTGTCACCTCCGAGGGGTCC   100198314   \n",
       "\n",
       "   start_coord_hg38  end_coord_hg38  Phenotype scores-ave_Rep1_Rep2  \\\n",
       "0         100198284       100198303                       -0.018881   \n",
       "1         100198285       100198304                        0.009292   \n",
       "2         100198291       100198310                        0.000734   \n",
       "3         100198292       100198311                       -0.008256   \n",
       "4         100198317       100198336                        0.001314   \n",
       "\n",
       "   Avg_LFC-Tfinal  z-score relative to NC condition Domain  CRISPick CutSite  \\\n",
       "0        0.035104               -0.889828      K562   Kox1         100198288   \n",
       "1        0.295731                0.580559      K562   Kox1         100198289   \n",
       "2        0.214023                0.119582      K562   Kox1         100198295   \n",
       "3        0.131427               -0.346402      K562   Kox1         100198296   \n",
       "4        0.165469               -0.154343      K562   Kox1         100198321   \n",
       "\n",
       "  Reference Sequence Strand of Target  TSS Position  \n",
       "0       NC_000012.12                +     100200816  \n",
       "1       NC_000012.12                +     100200816  \n",
       "2       NC_000012.12                +     100200816  \n",
       "3       NC_000012.12                +     100200816  \n",
       "4       NC_000012.12                +     100200816  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#reassign cut site for Nunez dataset guides\n",
    "NunezCleanDataCutSite = CRISPickCutSite(NunezCleanData,'strand' ,'hg38_coord', 'context_seq')\n",
    "NunezCleanData['CRISPick CutSite'] = NunezCleanDataCutSite\n",
    "NunezCleanData = pd.merge(NunezCleanData, CRISPick_geneListGilbertAndNunez,\n",
    "                          on= ['Target Gene Symbol'],how = 'inner')\n",
    "NunezCleanData.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "af89468d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>strand</th>\n",
       "      <th>sequence</th>\n",
       "      <th>context_seq</th>\n",
       "      <th>hg38_coord</th>\n",
       "      <th>start_coord_hg38</th>\n",
       "      <th>end_coord_hg38</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>CRISPick CutSite</th>\n",
       "      <th>Reference Sequence</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>Distance to TSS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTR6_+_100592059</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GCGAGCACATAGAGGGATCT</td>\n",
       "      <td>TGCTTCGAGCACATAGAGGGATCTGGGCCC</td>\n",
       "      <td>100198281</td>\n",
       "      <td>100198284</td>\n",
       "      <td>100198303</td>\n",
       "      <td>-0.018881</td>\n",
       "      <td>0.035104</td>\n",
       "      <td>-0.889828</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198288</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "      <td>-2528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACTR6_+_100592060</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GTCGAGCACATAGAGGGATC</td>\n",
       "      <td>CTGCTTCGAGCACATAGAGGGATCTGGGCC</td>\n",
       "      <td>100198282</td>\n",
       "      <td>100198285</td>\n",
       "      <td>100198304</td>\n",
       "      <td>0.009292</td>\n",
       "      <td>0.295731</td>\n",
       "      <td>0.580559</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198289</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "      <td>-2527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTR6_+_100592066</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GCCTGCTTCGAGCACATAGA</td>\n",
       "      <td>GTCCACCTGCTTCGAGCACATAGAGGGATC</td>\n",
       "      <td>100198288</td>\n",
       "      <td>100198291</td>\n",
       "      <td>100198310</td>\n",
       "      <td>0.000734</td>\n",
       "      <td>0.214023</td>\n",
       "      <td>0.119582</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198295</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "      <td>-2521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTR6_+_100592067</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GACCTGCTTCGAGCACATAG</td>\n",
       "      <td>GGTCCACCTGCTTCGAGCACATAGAGGGAT</td>\n",
       "      <td>100198289</td>\n",
       "      <td>100198292</td>\n",
       "      <td>100198311</td>\n",
       "      <td>-0.008256</td>\n",
       "      <td>0.131427</td>\n",
       "      <td>-0.346402</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198296</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "      <td>-2520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTR6_+_100592092</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GAGACTCTGTCACCTCCGAG</td>\n",
       "      <td>GGGCTAGACTCTGTCACCTCCGAGGGGTCC</td>\n",
       "      <td>100198314</td>\n",
       "      <td>100198317</td>\n",
       "      <td>100198336</td>\n",
       "      <td>0.001314</td>\n",
       "      <td>0.165469</td>\n",
       "      <td>-0.154343</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198321</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "      <td>-2495</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  ID Target Gene Symbol chromosome strand  \\\n",
       "0  ACTR6_+_100592059              ACTR6      chr12      +   \n",
       "1  ACTR6_+_100592060              ACTR6      chr12      +   \n",
       "2  ACTR6_+_100592066              ACTR6      chr12      +   \n",
       "3  ACTR6_+_100592067              ACTR6      chr12      +   \n",
       "4  ACTR6_+_100592092              ACTR6      chr12      +   \n",
       "\n",
       "               sequence                     context_seq  hg38_coord  \\\n",
       "0  GCGAGCACATAGAGGGATCT  TGCTTCGAGCACATAGAGGGATCTGGGCCC   100198281   \n",
       "1  GTCGAGCACATAGAGGGATC  CTGCTTCGAGCACATAGAGGGATCTGGGCC   100198282   \n",
       "2  GCCTGCTTCGAGCACATAGA  GTCCACCTGCTTCGAGCACATAGAGGGATC   100198288   \n",
       "3  GACCTGCTTCGAGCACATAG  GGTCCACCTGCTTCGAGCACATAGAGGGAT   100198289   \n",
       "4  GAGACTCTGTCACCTCCGAG  GGGCTAGACTCTGTCACCTCCGAGGGGTCC   100198314   \n",
       "\n",
       "   start_coord_hg38  end_coord_hg38  Phenotype scores-ave_Rep1_Rep2  \\\n",
       "0         100198284       100198303                       -0.018881   \n",
       "1         100198285       100198304                        0.009292   \n",
       "2         100198291       100198310                        0.000734   \n",
       "3         100198292       100198311                       -0.008256   \n",
       "4         100198317       100198336                        0.001314   \n",
       "\n",
       "   Avg_LFC-Tfinal  z-score relative to NC condition Domain  CRISPick CutSite  \\\n",
       "0        0.035104               -0.889828      K562   Kox1         100198288   \n",
       "1        0.295731                0.580559      K562   Kox1         100198289   \n",
       "2        0.214023                0.119582      K562   Kox1         100198295   \n",
       "3        0.131427               -0.346402      K562   Kox1         100198296   \n",
       "4        0.165469               -0.154343      K562   Kox1         100198321   \n",
       "\n",
       "  Reference Sequence Strand of Target  TSS Position  Distance to TSS  \n",
       "0       NC_000012.12                +     100200816            -2528  \n",
       "1       NC_000012.12                +     100200816            -2527  \n",
       "2       NC_000012.12                +     100200816            -2521  \n",
       "3       NC_000012.12                +     100200816            -2520  \n",
       "4       NC_000012.12                +     100200816            -2495  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#calculate guide distance from MANE Select annotated TSS \n",
    "NunezCleanData[\"Distance to TSS\"] = NunezCleanData.apply(\n",
    "    lambda r:(r[\"CRISPick CutSite\"] - r[\"TSS Position\"]) if r[\"Strand of Target\"] == \"+\" \\\n",
    "              else (r[\"TSS Position\"] - r[\"CRISPick CutSite\"]), axis=1)\n",
    "NunezCleanData = NunezCleanData[abs(NunezCleanData['Distance to TSS'])<=5000]\n",
    "NunezCleanData.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8884d31b",
   "metadata": {},
   "source": [
    "### Combine datasets together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "debfb94d",
   "metadata": {},
   "outputs": [],
   "source": [
    "GilbertRicinCleanData = deepcopy(GilbertRicinCleanData[['Target Gene Symbol','chromosome','sgRNA sequence', 'context seq', \n",
    "                                'CRISPick CutSite', 'Strand of Target', 'TSS Position', 'Distance to TSS',\n",
    "                                'zscoreRelativeNC_signed', 'Avg_LFC_signed', \n",
    "                                'condition', 'Domain']])\n",
    "\n",
    "\n",
    "GilbertRicinCleanData.rename(columns={\n",
    "                                      'sgRNA sequence':'sgRNA Sequence', \n",
    "                                      'context seq':'sgRNA Context Sequence',\n",
    "                                      'CRISPick CutSite': 'sgRNA \\'Cut\\' Position', \n",
    "                                      'Distance to TSS':'sgRNA \\'Cut\\' Site TSS Offset'\n",
    "                                     }, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "3986dca5",
   "metadata": {},
   "outputs": [],
   "source": [
    "NunezCleanData = deepcopy(NunezCleanData[['Target Gene Symbol','chromosome','sequence', \n",
    "                                               'context_seq', 'CRISPick CutSite',\n",
    "                                                       'Strand of Target', 'TSS Position', 'Distance to TSS',\n",
    "                                                       'Phenotype scores-ave_Rep1_Rep2','Avg_LFC-Tfinal', \n",
    "                         'z-score relative to NC','condition','Domain']])\n",
    "\n",
    "NunezCleanData.rename(columns={'sequence':'sgRNA Sequence', \n",
    "                                    'context_seq':'sgRNA Context Sequence',\n",
    "                                   'CRISPick CutSite': 'sgRNA \\'Cut\\' Position', \n",
    "                                      'Distance to TSS':'sgRNA \\'Cut\\' Site TSS Offset'}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "435bce6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "InHouseCleanData = deepcopy(InHouseCleanData\\\n",
    "                                 [['Target Gene Symbol','chromosome','sgRNA Sequence', \n",
    "                                   'sgRNA Context Sequence','sgRNA \\'Cut\\' Position',\n",
    "                                   'Strand of Target', 'TSS Position', 'sgRNA \\'Cut\\' Site TSS Offset',\n",
    "                                                       'z_scored_avg_lfc',\n",
    "                                                              'condition','Domain']])\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "6e55e0dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "NunezCleanData['DataSet'] = 'Nunez'\n",
    "GilbertRicinCleanData['DataSet'] = 'Gilbert'\n",
    "InHouseCleanData['DataSet'] = 'InHouse'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "352e90f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>z_scored_avg_lfc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>GCGAGCACATAGAGGGATCT</td>\n",
       "      <td>TGCTTCGAGCACATAGAGGGATCTGGGCCC</td>\n",
       "      <td>100198288.0</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816.0</td>\n",
       "      <td>-2528.0</td>\n",
       "      <td>-0.018881</td>\n",
       "      <td>0.035104</td>\n",
       "      <td>-0.889828</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>GTCGAGCACATAGAGGGATC</td>\n",
       "      <td>CTGCTTCGAGCACATAGAGGGATCTGGGCC</td>\n",
       "      <td>100198289.0</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816.0</td>\n",
       "      <td>-2527.0</td>\n",
       "      <td>0.009292</td>\n",
       "      <td>0.295731</td>\n",
       "      <td>0.580559</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>GCCTGCTTCGAGCACATAGA</td>\n",
       "      <td>GTCCACCTGCTTCGAGCACATAGAGGGATC</td>\n",
       "      <td>100198295.0</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816.0</td>\n",
       "      <td>-2521.0</td>\n",
       "      <td>0.000734</td>\n",
       "      <td>0.214023</td>\n",
       "      <td>0.119582</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>GACCTGCTTCGAGCACATAG</td>\n",
       "      <td>GGTCCACCTGCTTCGAGCACATAGAGGGAT</td>\n",
       "      <td>100198296.0</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816.0</td>\n",
       "      <td>-2520.0</td>\n",
       "      <td>-0.008256</td>\n",
       "      <td>0.131427</td>\n",
       "      <td>-0.346402</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>GAGACTCTGTCACCTCCGAG</td>\n",
       "      <td>GGGCTAGACTCTGTCACCTCCGAGGGGTCC</td>\n",
       "      <td>100198321.0</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816.0</td>\n",
       "      <td>-2495.0</td>\n",
       "      <td>0.001314</td>\n",
       "      <td>0.165469</td>\n",
       "      <td>-0.154343</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Target Gene Symbol chromosome        sgRNA Sequence  \\\n",
       "0              ACTR6      chr12  GCGAGCACATAGAGGGATCT   \n",
       "1              ACTR6      chr12  GTCGAGCACATAGAGGGATC   \n",
       "2              ACTR6      chr12  GCCTGCTTCGAGCACATAGA   \n",
       "3              ACTR6      chr12  GACCTGCTTCGAGCACATAG   \n",
       "4              ACTR6      chr12  GAGACTCTGTCACCTCCGAG   \n",
       "\n",
       "           sgRNA Context Sequence  sgRNA 'Cut' Position Strand of Target  \\\n",
       "0  TGCTTCGAGCACATAGAGGGATCTGGGCCC           100198288.0                +   \n",
       "1  CTGCTTCGAGCACATAGAGGGATCTGGGCC           100198289.0                +   \n",
       "2  GTCCACCTGCTTCGAGCACATAGAGGGATC           100198295.0                +   \n",
       "3  GGTCCACCTGCTTCGAGCACATAGAGGGAT           100198296.0                +   \n",
       "4  GGGCTAGACTCTGTCACCTCCGAGGGGTCC           100198321.0                +   \n",
       "\n",
       "   TSS Position  sgRNA 'Cut' Site TSS Offset  Phenotype scores-ave_Rep1_Rep2  \\\n",
       "0   100200816.0                      -2528.0                       -0.018881   \n",
       "1   100200816.0                      -2527.0                        0.009292   \n",
       "2   100200816.0                      -2521.0                        0.000734   \n",
       "3   100200816.0                      -2520.0                       -0.008256   \n",
       "4   100200816.0                      -2495.0                        0.001314   \n",
       "\n",
       "   Avg_LFC-Tfinal  z-score relative to NC condition Domain DataSet  \\\n",
       "0        0.035104               -0.889828      K562   Kox1   Nunez   \n",
       "1        0.295731                0.580559      K562   Kox1   Nunez   \n",
       "2        0.214023                0.119582      K562   Kox1   Nunez   \n",
       "3        0.131427               -0.346402      K562   Kox1   Nunez   \n",
       "4        0.165469               -0.154343      K562   Kox1   Nunez   \n",
       "\n",
       "   zscoreRelativeNC_signed  Avg_LFC_signed  z_scored_avg_lfc  \n",
       "0                      NaN             NaN               NaN  \n",
       "1                      NaN             NaN               NaN  \n",
       "2                      NaN             NaN               NaN  \n",
       "3                      NaN             NaN               NaN  \n",
       "4                      NaN             NaN               NaN  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DatasetCombine = pd.concat([NunezCleanData, GilbertRicinCleanData, InHouseCleanData]).reset_index(drop = True)\n",
    "DatasetCombine.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "d7f4b994-3c59-4e96-8ff6-a7db8580375b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    204895\n",
       "2      9728\n",
       "3       364\n",
       "Name: DataSet, dtype: int64"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#how many context sequences appear across multiple datasets?\n",
    "DatasetCombine.groupby('sgRNA Context Sequence')['DataSet'].nunique().value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "fcf7adda-950c-4335-af98-e15df1a25bf4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>DataSet</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3047</th>\n",
       "      <td>AURKB</td>\n",
       "      <td>AATTTGGAATTAATGACTAAATTGAGGTAA</td>\n",
       "      <td>-</td>\n",
       "      <td>826.0</td>\n",
       "      <td>Nunez</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3048</th>\n",
       "      <td>AURKB</td>\n",
       "      <td>CCAGGGCAAGTCTAAATTACAATTTGGAAT</td>\n",
       "      <td>-</td>\n",
       "      <td>806.0</td>\n",
       "      <td>Nunez</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3049</th>\n",
       "      <td>AURKB</td>\n",
       "      <td>AAGGATATAAAGTTACAGTTTAAAAGGCCC</td>\n",
       "      <td>-</td>\n",
       "      <td>778.0</td>\n",
       "      <td>Nunez</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3050</th>\n",
       "      <td>AURKB</td>\n",
       "      <td>ATGATATCTGGGACTGAGCAGGCAGGGCCT</td>\n",
       "      <td>-</td>\n",
       "      <td>674.0</td>\n",
       "      <td>Nunez</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3051</th>\n",
       "      <td>AURKB</td>\n",
       "      <td>AATGATATCTGGGACTGAGCAGGCAGGGCC</td>\n",
       "      <td>-</td>\n",
       "      <td>673.0</td>\n",
       "      <td>Nunez</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Target Gene Symbol          sgRNA Context Sequence Strand of Target  \\\n",
       "3047              AURKB  AATTTGGAATTAATGACTAAATTGAGGTAA                -   \n",
       "3048              AURKB  CCAGGGCAAGTCTAAATTACAATTTGGAAT                -   \n",
       "3049              AURKB  AAGGATATAAAGTTACAGTTTAAAAGGCCC                -   \n",
       "3050              AURKB  ATGATATCTGGGACTGAGCAGGCAGGGCCT                -   \n",
       "3051              AURKB  AATGATATCTGGGACTGAGCAGGCAGGGCC                -   \n",
       "\n",
       "      sgRNA 'Cut' Site TSS Offset DataSet  \n",
       "3047                        826.0   Nunez  \n",
       "3048                        806.0   Nunez  \n",
       "3049                        778.0   Nunez  \n",
       "3050                        674.0   Nunez  \n",
       "3051                        673.0   Nunez  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# looking specifically context sequences present in both in-house and other datasets to verify that cut position and cut site offset\n",
    "    #were calculated correctly\n",
    "datasets_per_seq = DatasetCombine.groupby('sgRNA Context Sequence')['DataSet'].nunique()\n",
    "multi_dataset_seqs = datasets_per_seq.index[datasets_per_seq > 1]\n",
    "multi_dataset_seq_df = DatasetCombine[DatasetCombine['sgRNA Context Sequence'].isin(multi_dataset_seqs)]\n",
    "multi_dataset_seq_df = multi_dataset_seq_df[['Target Gene Symbol','sgRNA Context Sequence', 'Strand of Target','sgRNA \\'Cut\\' Site TSS Offset', 'DataSet']].drop_duplicates()\n",
    "multi_dataset_seq_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "285a1639-70a8-4173-b384-180737fb575d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>DataSet</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>Gilbert</th>\n",
       "      <th>InHouse</th>\n",
       "      <th>Nunez</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAAAAAAATTCTGCAGGAGGCAATAGGTTT</td>\n",
       "      <td>DDX18</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>935.0</td>\n",
       "      <td>935.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAAAAAACGTTAAAGAAATTGTAAAGGAGA</td>\n",
       "      <td>DNTTIP2</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-561.0</td>\n",
       "      <td>-562.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAAAATATATGCCATTTACCCCTCGGGAA</td>\n",
       "      <td>HEATR1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-554.0</td>\n",
       "      <td>-555.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAAAATTCCTGGAAAACCCAGGTAGGACT</td>\n",
       "      <td>METTL17</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>617.0</td>\n",
       "      <td>617.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAAAATTGAGGTCCTGAGTTAAGTGGCTG</td>\n",
       "      <td>MRPS25</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-423.0</td>\n",
       "      <td>-424.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11159</th>\n",
       "      <td>TTTTTCATTGTTAAACCTGAAGACGGGGCA</td>\n",
       "      <td>TTK</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>493.0</td>\n",
       "      <td>493.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11160</th>\n",
       "      <td>TTTTTGGGCAGCACGGCTCCCATTCGGAAC</td>\n",
       "      <td>DNAJA3</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-887.0</td>\n",
       "      <td>-887.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11161</th>\n",
       "      <td>TTTTTGTCGTAATCCTATTGAGTTCGGGTC</td>\n",
       "      <td>SNRPF</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-752.0</td>\n",
       "      <td>-752.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11162</th>\n",
       "      <td>TTTTTTGCTCGCAGAAACAATACGTGGTCA</td>\n",
       "      <td>POP5</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-535.0</td>\n",
       "      <td>-536.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11163</th>\n",
       "      <td>TTTTTTGTATCCGTGTTGGTTCTTAGGCGG</td>\n",
       "      <td>RPL4</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-58.0</td>\n",
       "      <td>-59.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11164 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "DataSet          sgRNA Context Sequence Target Gene Symbol Strand of Target  \\\n",
       "0        AAAAAAAATTCTGCAGGAGGCAATAGGTTT              DDX18                +   \n",
       "1        AAAAAAACGTTAAAGAAATTGTAAAGGAGA            DNTTIP2                -   \n",
       "2        AAAAAATATATGCCATTTACCCCTCGGGAA             HEATR1                -   \n",
       "3        AAAAAATTCCTGGAAAACCCAGGTAGGACT            METTL17                +   \n",
       "4        AAAAAATTGAGGTCCTGAGTTAAGTGGCTG             MRPS25                -   \n",
       "...                                 ...                ...              ...   \n",
       "11159    TTTTTCATTGTTAAACCTGAAGACGGGGCA                TTK                +   \n",
       "11160    TTTTTGGGCAGCACGGCTCCCATTCGGAAC             DNAJA3                +   \n",
       "11161    TTTTTGTCGTAATCCTATTGAGTTCGGGTC              SNRPF                +   \n",
       "11162    TTTTTTGCTCGCAGAAACAATACGTGGTCA               POP5                -   \n",
       "11163    TTTTTTGTATCCGTGTTGGTTCTTAGGCGG               RPL4                -   \n",
       "\n",
       "DataSet  Gilbert  InHouse  Nunez  \n",
       "0            NaN    935.0  935.0  \n",
       "1            NaN   -561.0 -562.0  \n",
       "2            NaN   -554.0 -555.0  \n",
       "3            NaN    617.0  617.0  \n",
       "4            NaN   -423.0 -424.0  \n",
       "...          ...      ...    ...  \n",
       "11159        NaN    493.0  493.0  \n",
       "11160        NaN   -887.0 -887.0  \n",
       "11161        NaN   -752.0 -752.0  \n",
       "11162        NaN   -535.0 -536.0  \n",
       "11163        NaN    -58.0  -59.0  \n",
       "\n",
       "[11164 rows x 6 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pivoted_df = multi_dataset_seq_df.pivot(index = ['sgRNA Context Sequence','Target Gene Symbol','Strand of Target'],\n",
    "                                        columns = 'DataSet',\n",
    "                                        values = 'sgRNA \\'Cut\\' Site TSS Offset').reset_index()\n",
    "pivoted_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "13c6990b-b6ac-4663-9831-b703112b9f46",
   "metadata": {},
   "source": [
    "It appears at first glance that the calculated TSS offset for the Nunez dataset is one off from the correct TSS offset for the in-house dataset (that was derived from CRISPick) ONLY when the guide is on the negative strand. Check whether this hypothesis is correct."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "bbef3c6e-e730-476a-82ee-cf97b734e579",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Strand of Target  InHouse - Nunez  InHouse - Gilbert\n",
       "+                 0.0              0.0                  277\n",
       "-                 1.0              1.0                   87\n",
       "dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pivoted_df['InHouse - Nunez'] = pivoted_df['InHouse'] - pivoted_df['Nunez']\n",
    "pivoted_df['InHouse - Gilbert'] = pivoted_df['InHouse'] - pivoted_df['Gilbert']\n",
    "pivoted_df[['Strand of Target','InHouse - Nunez','InHouse - Gilbert']].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aee47a5f-7458-4ad3-9f6b-cc5e61ad820f",
   "metadata": {},
   "source": [
    "The calculated cut site is indeed one off for negative strand targeting guides."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "66b10ee0-e28e-4e00-95d0-1a6c722938ab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>z_scored_avg_lfc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>47174</th>\n",
       "      <td>MRPS25</td>\n",
       "      <td>chr3</td>\n",
       "      <td>GATTGAGGTCCTGAGTTAAG</td>\n",
       "      <td>AAAAAATTGAGGTCCTGAGTTAAGTGGCTG</td>\n",
       "      <td>15065739.0</td>\n",
       "      <td>-</td>\n",
       "      <td>15065315.0</td>\n",
       "      <td>-423.0</td>\n",
       "      <td>-0.019317</td>\n",
       "      <td>0.026164</td>\n",
       "      <td>-0.940264</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142713</th>\n",
       "      <td>MRPS25</td>\n",
       "      <td>3.0</td>\n",
       "      <td>AATTGAGGTCCTGAGTTAAG</td>\n",
       "      <td>AAAAAATTGAGGTCCTGAGTTAAGTGGCTG</td>\n",
       "      <td>15065739.0</td>\n",
       "      <td>-</td>\n",
       "      <td>15065315.0</td>\n",
       "      <td>-423.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.402571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142714</th>\n",
       "      <td>MRPS25</td>\n",
       "      <td>3.0</td>\n",
       "      <td>AATTGAGGTCCTGAGTTAAG</td>\n",
       "      <td>AAAAAATTGAGGTCCTGAGTTAAGTGGCTG</td>\n",
       "      <td>15065739.0</td>\n",
       "      <td>-</td>\n",
       "      <td>15065315.0</td>\n",
       "      <td>-423.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.761167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142715</th>\n",
       "      <td>MRPS25</td>\n",
       "      <td>3.0</td>\n",
       "      <td>AATTGAGGTCCTGAGTTAAG</td>\n",
       "      <td>AAAAAATTGAGGTCCTGAGTTAAGTGGCTG</td>\n",
       "      <td>15065739.0</td>\n",
       "      <td>-</td>\n",
       "      <td>15065315.0</td>\n",
       "      <td>-423.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.075731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142716</th>\n",
       "      <td>MRPS25</td>\n",
       "      <td>3.0</td>\n",
       "      <td>AATTGAGGTCCTGAGTTAAG</td>\n",
       "      <td>AAAAAATTGAGGTCCTGAGTTAAGTGGCTG</td>\n",
       "      <td>15065739.0</td>\n",
       "      <td>-</td>\n",
       "      <td>15065315.0</td>\n",
       "      <td>-423.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.764584</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Target Gene Symbol chromosome        sgRNA Sequence  \\\n",
       "47174              MRPS25       chr3  GATTGAGGTCCTGAGTTAAG   \n",
       "142713             MRPS25        3.0  AATTGAGGTCCTGAGTTAAG   \n",
       "142714             MRPS25        3.0  AATTGAGGTCCTGAGTTAAG   \n",
       "142715             MRPS25        3.0  AATTGAGGTCCTGAGTTAAG   \n",
       "142716             MRPS25        3.0  AATTGAGGTCCTGAGTTAAG   \n",
       "\n",
       "                sgRNA Context Sequence  sgRNA 'Cut' Position Strand of Target  \\\n",
       "47174   AAAAAATTGAGGTCCTGAGTTAAGTGGCTG            15065739.0                -   \n",
       "142713  AAAAAATTGAGGTCCTGAGTTAAGTGGCTG            15065739.0                -   \n",
       "142714  AAAAAATTGAGGTCCTGAGTTAAGTGGCTG            15065739.0                -   \n",
       "142715  AAAAAATTGAGGTCCTGAGTTAAGTGGCTG            15065739.0                -   \n",
       "142716  AAAAAATTGAGGTCCTGAGTTAAGTGGCTG            15065739.0                -   \n",
       "\n",
       "        TSS Position  sgRNA 'Cut' Site TSS Offset  \\\n",
       "47174     15065315.0                       -423.0   \n",
       "142713    15065315.0                       -423.0   \n",
       "142714    15065315.0                       -423.0   \n",
       "142715    15065315.0                       -423.0   \n",
       "142716    15065315.0                       -423.0   \n",
       "\n",
       "        Phenotype scores-ave_Rep1_Rep2  Avg_LFC-Tfinal  \\\n",
       "47174                        -0.019317        0.026164   \n",
       "142713                             NaN             NaN   \n",
       "142714                             NaN             NaN   \n",
       "142715                             NaN             NaN   \n",
       "142716                             NaN             NaN   \n",
       "\n",
       "        z-score relative to NC condition Domain  DataSet  \\\n",
       "47174                -0.940264      K562   Kox1    Nunez   \n",
       "142713                     NaN      A549   Kox1  InHouse   \n",
       "142714                     NaN      A549   Zim3  InHouse   \n",
       "142715                     NaN    HCT116   Kox1  InHouse   \n",
       "142716                     NaN    HCT116   Zim3  InHouse   \n",
       "\n",
       "        zscoreRelativeNC_signed  Avg_LFC_signed  z_scored_avg_lfc  \n",
       "47174                       NaN             NaN               NaN  \n",
       "142713                      NaN             NaN         -0.402571  \n",
       "142714                      NaN             NaN         -0.761167  \n",
       "142715                      NaN             NaN          0.075731  \n",
       "142716                      NaN             NaN         -0.764584  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#correct this offset\n",
    "DatasetCombine.loc[(DatasetCombine['DataSet'].isin(['Nunez', 'Gilbert'])) & (DatasetCombine['Strand of Target'] == '-'), \n",
    "                      'sgRNA \\'Cut\\' Site TSS Offset'] += 1\n",
    "\n",
    "#look at an example to verify whether it worked\n",
    "DatasetCombine[DatasetCombine['sgRNA Context Sequence'] == 'AAAAAATTGAGGTCCTGAGTTAAGTGGCTG']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5d59a6ce-1040-48d5-8e62-1d399d5ac279",
   "metadata": {},
   "source": [
    "### Remove overlapping genes\n",
    "Guide RNAs which may target protein coding genes other than their intended target have the potential to confound analyses. Since the majority of guides within the combined tiling datasets are located within 1kb of the MANE Select TSS of their target gene, we remove all genes with any GENCODE annotated TSS within 1kb of any other GENCODE annotated TSS corresponding to a different gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "99b4bc5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove overlapping genes\n",
    "overlapping_gene = pd.read_csv('../../Data/geneOverlap/GW_ensembl_protein_coding_df_1kb.csv')\n",
    "overlapping_gene_2_3 =overlapping_gene[overlapping_gene['overlap type'].isin([2,3])]\n",
    "gene_2_3_unique = set(list(overlapping_gene_2_3['Selected Gene name']) +\\\n",
    "list(overlapping_gene_2_3['Overlapping Gene name']))\n",
    "NewDatasetCombine = deepcopy(DatasetCombine[~DatasetCombine['Target Gene Symbol'].isin(gene_2_3_unique)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "f0556b80",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# did removing overlapping genes remove all multi-target guides?\n",
    "conseq_with_varied_gene = NewDatasetCombine.groupby('sgRNA Context Sequence').filter(lambda NewDatasetCombine: NewDatasetCombine['Target Gene Symbol'].nunique() > 1)\n",
    "#if yes, should be 0\n",
    "len(conseq_with_varied_gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "e97fd43f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "166624"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NewDatasetCombine['sgRNA Context Sequence'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "e2723bb2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "431095"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(NewDatasetCombine)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "9a2ddbb4-7f2a-4d4f-a4ee-32a316ae6c70",
   "metadata": {},
   "outputs": [],
   "source": [
    "#standardizes syntax with which chromosomes are labeled across datasets\n",
    "def standardize_chromosome(chromosome):\n",
    "    chromosome = str(chromosome).replace('.0', '')\n",
    "    if not chromosome.startswith('chr'):\n",
    "        chromosome = 'chr' + chromosome\n",
    "    return chromosome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "889b9a42",
   "metadata": {},
   "outputs": [],
   "source": [
    "NewDatasetCombine['chromosome'] = NewDatasetCombine['chromosome'].apply(standardize_chromosome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "c8b3227b-a94e-436f-9d4e-264a5999497c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#define list of essential genes in in-house datasets\n",
    "EssGene = ess_noness[ess_noness.Category == 'Essential']['Target Gene Symbol'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "e6fac2b8-4e83-4235-92b1-5cd9323696cb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DataSet  condition  Domain\n",
       "Nunez    K562       Kox1      68575\n",
       "InHouse  A549       Zim3      42609\n",
       "         HCT116     Kox1      42609\n",
       "         A549       Kox1      42570\n",
       "         HCT116     Zim3      42544\n",
       "Gilbert  K562       Kox1      18509\n",
       "dtype: int64"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#remove non-essential genes\n",
    "InhouseEss = NewDatasetCombine[(NewDatasetCombine.DataSet == 'InHouse') & (NewDatasetCombine['Target Gene Symbol'].isin(EssGene))]\n",
    "other = NewDatasetCombine[~(NewDatasetCombine.DataSet == 'InHouse')]\n",
    "newData = pd.concat([InhouseEss,other])\n",
    "newData.value_counts(['DataSet', 'condition', 'Domain'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "3251a8cc-fe56-4e99-acef-736f73f2fcb2",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_path = '../../Data/cleanedData/'\n",
    "newData.to_csv(out_path + 'CombinedCleanedDatasets.csv', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6f2f7b0a-9fc7-4e24-b2b6-0b2bc8c265c6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
